### PR TITLE
[G2M] data key exploration and chart-system (plotly line chart) stories

### PR DESCRIPTION
**Pull request by woozyking**
line charts

### CODE REVIEW

**Comment by vxsl** [comment]
nice :+1: 

the `Explore:` text input loses focus everytime I type a character because it is momentarily replaced with `Loading...` and then re-rendered. Makes it difficult to change the input.

**Review by vxsl — APPROVED** [review]

**Comment by woozyking** [comment]
> the Explore: text input loses focus everytime I type a character because it is momentarily replaced with Loading... and then re-rendered. Makes it difficult to change the input.

oh ye that was with the vanilla input. #3 uses lumen labs components so should be a bit more intuitive

File: src/index.js [hunk 1]
```diff
@@ -43,21 +56,49 @@ export const useSample = (sample) => {
...
+      setKeys(buildKeys(data))
+    } catch(e) {
+      // TODO: handle error
+      setKeys([])
+    }
+  }, [data])
+  // set parsed keys by inferred value type, react to keys or data change
+  const [typedKeys, setTypedKeys] = useState({})
+  useEffect(() => {
+    if (keys && keys.length > 0 && data && data.length > 0) {
```
Comment (hyx131) [inline src/index.js]: or `if (keys?.length && data?.length)` 🌚

**Review by hyx131 — APPROVED** [review]
looks good!

File: src/index.js (hunk 1 above)
Comment (woozyking) [reply src/index.js]: revised in [c789f4f](https://github.com/EQWorks/api-explorer/pull/2/commits/c789f4fc2317940e14e22184e5d53e55402be738) along some other logic simplification/cleanup

**Pull request merged**